### PR TITLE
update/regex

### PR DESCRIPTION
**Pull request by BarbUk**
- **regex: Disallow terminating colon in URLs**
- **Match all regex**

REGEX_URL_AS_IS
REGEX_URL_HTTP
REGEX_URL_FILE
REGEX_URL_VOIP
REGEX_EMAIL
REGEX_NEWS_MAN

### CODE REVIEW

File: termite.cc [hunk 1]
```diff
@@ -44,6 +44,21 @@
...
+static const TerminalRegexPattern url_regex_patterns[] = {
+  REGEX_URL_AS_IS,
+  REGEX_URL_HTTP,
+  REGEX_URL_FILE,
+  REGEX_URL_VOIP,
+  REGEX_EMAIL,
+  REGEX_NEWS_MAN,
+};
+
+static guint n_url_regexes;
```
Comment (aperezdc) [inline termite.cc:60]: This is not needed, we can use `G_N_ELEMENTS()` in the loop below.

```suggestion
```

File: termite.cc [hunk 2]
```diff
@@ -44,6 +44,21 @@
 
 #include "url_regex.hh"
 
+typedef struct {
+  const char *pattern;
+} TerminalRegexPattern;
+
+static const TerminalRegexPattern url_regex_patterns[] = {
```
Comment (aperezdc) [inline termite.cc:51]: Any reason why this cannot be an array of `const char*`? Something like this:

```suggestion
static const char *const url_regex_patterns[] = {
```

File: termite.cc [hunk 3]
```diff
@@ -1648,12 +1663,16 @@ static void set_config(GtkWindow *window, VteTerminal *vte, GtkWidget *scrollbar
...
 
     if (info->clickable_url) {
-        info->tag = vte_terminal_match_add_regex(vte,
-                vte_regex_new_for_match(url_regex,
-                                        (gssize) strlen(url_regex),
-                                        PCRE2_MULTILINE | PCRE2_NOTEMPTY,
-                                        nullptr),
+        n_url_regexes = G_N_ELEMENTS (url_regex_patterns);
+        for (int i = 0; i < n_url_regexes; ++i)
+        {
```
Comment (aperezdc) [inline termite.cc:1668]: Just use `G_N_ELEMENTS()` directly in the loop condition, the compiler is smart enough to figure out that the macro expands to a constant value:

```suggestion
        for (int i = 0; i < G_N_ELEMENTS(url_regex_patterns); ++i) {
```

**Review by aperezdc — CHANGES_REQUESTED** [review]
Thanks for the patch! I do have a couple of comments to make it a bit nicer, could you please take a look and update it accordingly? 🙏🏼

File: termite.cc (hunk 2 above)
Comment (BarbUk) [reply termite.cc:51]: no valid reason other than my lack of competence in cpp

File: termite.cc (hunk 1 above)
Comment (BarbUk) [reply termite.cc:60]: yes, thanks

File: termite.cc (hunk 3 above)
Comment (BarbUk) [reply termite.cc:1668]: yes, thanks.

**Comment by BarbUk** [comment]
> I do have a couple of comments to make it a bit nicer, could you please take a look and update it accordingly? 🙏🏼

Thanks for the review.
I have updated the PR.

**Review by aperezdc — APPROVED** [review]
@BarbUk Nicer now, thanks for taking care of updating the PR. Let's merge this 🚢

**Pull request merged**